File: Arise.FileSyncer.Service/Ipc/Messages/DeleteProfileMessage.cs
using System;

namespace Arise.FileSyncer.Service.Ipc.Messages
{
    public class DeleteProfileMessage : IpcMessage
    {
        public override string Command => "deleteProfile";

        public Guid ProfileId { get; set; }

        internal override void Process(IpcController ipc)
        {
            bool success = ipc.Service.Peer.Profiles.RemoveProfile(ProfileId);
            ipc.Send(new DeleteProfileResultMessage() { Success = success });
        }
    }
}

File: Arise.FileSyncer.ElectronUI/src/main/ipc/messages/deleteProfileMessage.ts
import { IpcMessage } from "../ipcMessage";

export const Command = "deleteProfile";

export interface Message extends IpcMessage {
    ProfileId: string,
}

export function Create(profileId: string): Message {
    let message: Message = {
        Command: Command,
        ProfileId: profileId,
    };

    return message;
}